api/src/logger.ts

```diff
@@ -10,4 +19,25 @@ if (env.LOG_STYLE !== 'raw') {
 
 const logger = pino(pinoOptions);
 
+export const expressLogger = pinoHTTP({
+	logger,
+	serializers: {
+		req(request: Request) {
+			const output = stdSerializers.req(request);
+			output.url = redactQuery(output.url);
+			return output;
+		},
+	},
+}) as RequestHandler;
+
 export default logger;
+
+function redactQuery(originalPath: string) {
+	const url = new URL(originalPath, 'http://example.com/');
+
+	if (url.searchParams.has('access_token')) {
```

Comment:
So much easier to redact Node Logs!

What if the redacted options are dynamic?
```shell
LOG_LEVEL="debug"
LOG_REDACT_REQUEST="req.headers.authorization,req.cookies.directus_refresh_token"
LOG_REDACT_PARAMS="access_token"
LOG_STYLE="pretty"
```

Reply:
> What if the redacted options are dynamic?

I was thinking about that too. For now, this is definitely the safe way to go. 

If the specific need to write the sensitive tokens comes up with a well defined use case, we can discuss if it's 80/20 and worth implementing 👍🏻

Reply:
> I was thinking about that too. For now, this is definitely the safe way to go.

Awesome! That works.
For now testing locally.

Reply:
Are the PUT request logs deleted and masked as `You don't have permission to access this.` in the log?

Reply:
There's no explicit masking for particular requests or methods, no. I'm assuming that's just because your user didn't have permission (also it's PATCH, not PUT) 👍🏻

Reply:
Yeah PATCH, sorry.

> I'm assuming that's just because your user didn't have permission

Hmmm interesting...Primary admin user updating self.

Couldn't find logged activity for the update...but it is in Directus Activity, so that is cool!